agent/util-scripts/pbench-move-results

```diff
@@ -172,120 +186,84 @@ for dir in `/bin/ls -ort -d */ | awk '{print $8}' | grep -v "^tools-" | grep -v
         /bin/cp pbench.log $pbench_run_name/
     fi
 
+    # if -u was specified, store the specified user in metadata.log
+    if [ ! -z $user ] ;then
+	mdlog=${pbench_run_name}/metadata.log
+	echo $user | pbench-add-metalog-option ${mdlog} run user
+    fi
+
+    # if -p was specified, store the specified prefix in metadata.log
+    if [ ! -z $prefix ] ;then
+	mdlog=${pbench_run_name}/metadata.log
+	echo $prefix | pbench-add-metalog-option ${mdlog} run prefix
+    fi
+
     results_size=`du -sm $pbench_run_name | awk '{print $1}'`
     debug_log "preparing to copy $results_size MB of data from $pbench_run/$pbench_run_name"
 
-    tarball="$pbench_run_name.tar.xz"
+    # Create a temp directory $tmp/$controller to contain the tarball
+    # and the md5 file (as ${tb}.tar.xz.md5.check). Copy the directory
+    # with scp -r $tmp/$controller $remote: that will create the
+    # $controller subdirectory on the remote (if necessary) OR fail.
+
+    # If it does not fail, then check the MD5 sum and rename the foo.tar.xz.md5.check file
+    # to foo.tar.xz.md5. That's the signal that the agent has finished with this tarball.
+
+    tarball="$tmp/$controller/$pbench_run_name.tar.xz"
     if [[ ${xz_single_threaded} != "1" ]] ;then
 	echo "tar --create --force-local \"$pbench_run_name\" | xz -T0 > \"$tarball\" "
 	tar --create --force-local "$pbench_run_name" | xz -T0 > "$tarball"
     else
 	echo "tar --create --xz --force-local --file=\"$tarball\" \"$pbench_run_name\" "
-	tar --create --xz --force-local --file="$tarball" "$pbench_run_name" 
+	tar --create --xz --force-local --file="$tarball" "$pbench_run_name"
     fi
-    
+
     if [ $? -ne 0 ]; then
         error_log "ERROR: tar failed for $pbench_run/$pbench_run_name, skipping"
         rm -f "$tarball"
         let failures=failures+1
         continue
     fi
-    md5sum "$tarball" > "$tarball.md5"
+
+    tarballmd5="$tmp/$controller/$tarball.md5"
```

Comment:
This should be `tarballmd5="$tmp/$controller/${tarball##*/}.md5"` otherwise it is creating wrong URL of the md5.